frmts/tiledb/tiledbsparse.cpp

```diff
@@ -0,0 +1,4934 @@
+{
+    if (m_oLayerExtent.IsInit())
+    {
+        *psExtent = m_oLayerExtent;
+        return OGRERR_NONE;
+    }
+    return OGRLayer::GetExtent(psExtent, bForce);
+}
+
+/************************************************************************/
+/*                         ResetReading()                               */
+/************************************************************************/
+
+void OGRTileDBLayer::ResetReading()
+{
+    if (m_eCurrentMode == CurrentMode::WriteInProgress && !m_array)
+        return;
+
+    SwitchToReadingMode();
+    ResetBuffers();
+    m_nNextFID = 1;
+    m_nOffsetInResultSet = 0;
+    m_nRowCountInResultSet = 0;
+    m_query.reset();
+    m_bQueryComplete = false;
+}
+
+/************************************************************************/
+/*                         CreateField()                                */
+/************************************************************************/
+
+OGRErr OGRTileDBLayer::CreateField(OGRFieldDefn *poField, int /* bApproxOK*/)
+{
+    if (!m_bUpdatable)
+        return OGRERR_FAILURE;
+    if (m_schema)
+    {
+        CPLError(CE_Failure, CPLE_NotSupported,
+                 "Cannot add field after schema has been initialized");
+        return OGRERR_FAILURE;
+    }
+    if (poField->GetType() == OFTStringList)
+    {
+        CPLError(CE_Failure, CPLE_NotSupported, "Unsupported field type");
+        return OGRERR_FAILURE;
+    }
+    const char *pszFieldName = poField->GetNameRef();
+    if (m_poFeatureDefn->GetFieldIndex(pszFieldName) >= 0 ||
+        pszFieldName == m_osFIDColumn ||
+        strcmp(pszFieldName, GetGeometryColumn()) == 0 ||
+        pszFieldName == m_osXDim || pszFieldName == m_osYDim ||
+        pszFieldName == m_osZDim)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "A field or dimension of same name (%s) already exists",
+                 pszFieldName);
+        return OGRERR_FAILURE;
+    }
+    OGRFieldDefn oFieldDefn(poField);
+    m_poFeatureDefn->AddFieldDefn(&oFieldDefn);
+    return OGRERR_NONE;
+}
+
+/************************************************************************/
+/*                      InitializeSchemaAndArray()                      */
+/************************************************************************/
+
+void OGRTileDBLayer::InitializeSchemaAndArray()
+{
+    m_bInitializationAttempted = true;
+
+    try
+    {
+        // create the tiledb schema
+        // dimensions will be _x and _y, we can also add _z (2.5d)
+        // filters should be implemented as per tiledbdense
+        // set dimensions and attribute type for schema
+        // we will use row-major for now but we could use hilbert indexing
+        m_schema.reset(new tiledb::ArraySchema(*m_ctx, TILEDB_SPARSE));
+        m_schema->set_tile_order(TILEDB_ROW_MAJOR);
+        m_schema->set_cell_order(TILEDB_ROW_MAJOR);
+
+        m_schema->set_coords_filter_list(*m_filterList);
```

Comment:
Need to add `m_schema→set_offsets_filter_list(*m_filterList)` so that variable length offsets are also compressed. Additionally we need to set tile capacity as per per https://github.com/PDAL/PDAL/blob/master/plugins/tiledb/io/TileDBWriter.cpp#L620 100000 is small though, it can be 500000.

Reply:
done